pkg/provision/engine.go

```diff
@@ -732,6 +827,18 @@ func (e *NativeEngine) installDeployment(ctx context.Context, getter gridtypes.W
 	}
 }
 
+func (e *NativeEngine) lockDeployment(ctx context.Context, getter gridtypes.WorkloadGetter, lock bool) {
+	for _, typ := range e.order {
```

Comment:
Does the order matter here depending on whether it's locking or unlocking? So if it's unlocking the order should be reversed as in the uninstallDeployment method?

Reply:
Yeah, you have a point my plan was actually to make it lock, and unlock in the same order as we do deprovision and provision. I think i was carried away with other changes